versions/3.0.4.md

```diff
@@ -1044,7 +1044,7 @@ Field Name | Type | Description
 <a name="parameterDescription"></a>description | `string` | A brief description of the parameter. This could contain examples of use. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
 <a name="parameterRequired"></a>required | `boolean` | Determines whether this parameter is mandatory. If the [parameter location](#parameterIn) is `"path"`, this property is **REQUIRED** and its value MUST be `true`. Otherwise, the property MAY be included and its default value is `false`.
 <a name="parameterDeprecated"></a> deprecated | `boolean` | Specifies that a parameter is deprecated and SHOULD be transitioned out of usage. Default value is `false`.
-<a name="parameterAllowEmptyValue"></a> allowEmptyValue | `boolean` | Sets the ability to pass empty-valued parameters. This is valid only for `query` parameters and allows sending a parameter with an empty value. Default value is `false`. If [`style`](#parameterStyle) is used, and if behavior is `n/a` (cannot be serialized), the value of `allowEmptyValue` SHALL be ignored. Use of this property is NOT RECOMMENDED, as it is likely to be removed in a later revision.
+<a name="parameterAllowEmptyValue"></a> allowEmptyValue | `boolean` | If `true`, allows passing empty-valued parameters regardless of `schema` or `content`. This is valid only for `query` parameters. Default value is `false`. If [`style`](#parameterStyle) is used, and if behavior is `n/a` (cannot be serialized), the value of `allowEmptyValue` SHALL be ignored. Use of this property is NOT RECOMMENDED, as it is likely to be removed in a later revision; instead, ensure that the parameter's schema accepts any desired empty value(s).
```

Comment:
I'd say both have empty values because if I run both variants through the standard URL parser of my current favorite programming language or build a service with my favorite  web app framework both forms result in 
~~~js
{ 'foo' => '' }
~~~
so my server-side code would not be able to spot the difference.

Which of course is only my very personal opinion 😄

Reply:
@mikekistler it's not possible to produce `?foo` (without an `=`).  This is unfortunate, but it was discussed extensively in issue #1573 (the previous attempt to sort this out).

Reply:
> it's not possible to produce `?foo` (without an `=`)

Just did with
~~~js
http.get("http://localhost:3000/?foo", ...
~~~

My minimalistic server tells me that it received the URL `/?foo`. No `=` anywhere.

Reply:
I think Henry means it's not possible to produce using an openapi tool to do the serialization? (Originally it was thought that `allowEmptyValue` would do this, but then in #1573 it was observed that this doesn't fit in with the URI Template spec.)

Reply:
@ralfhandl – @karenetheridge is correct, and per #1573 whatever tool did that is non-compliant.

Reply:
> Consensus on the TSC is that this feature has ambiguous meanings, and interacts in awkward ways with Schema Object and Parameter `style`.

Definitely agree with that 😁